[Amresh] iP

amresh2000 opened the pull request:
## Duke
> Hard work beats talent, when talent does not work hard
####Uses of Duke
- Organize your day
- Add todo tasks 
- Live updating of list

#### How to Use
1. Download it from [here] (https://github.com/nus-cs2103-AY2223S1/ip.git) 
2. double-click it
3. add ~~your~~ *all your* tasks
4. let it manage your tasks for you 😄 

#### Sample code of Duke
Here is the `main` method
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

#### Features Completed
- [x] Chatbot with basic todo, event, deadline
- [x] Reminder System
- [x] Analytics

Commit:
Add Gradle support

amresh2000 committed:
Level 1. Greet, Echo, Exit

amresh2000 committed:
Level 2. Add, List

amresh2000 committed:
Level 3. Mark as Done
Added new Class Task

amresh2000 committed:
Level 4. Added new Classes ToDos, Events, Deadlines

amresh2000 committed:
Level 5. Handle Errors. Added DukeException class

amresh2000 committed:
Level 6. Delete

amresh2000 committed:
Level 7. Save the tasks in hard drive

amresh2000 committed:
Level 8. Dates and Times

amresh2000 committed:
Merge branch 'branch-Level-7'

amresh2000 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

amresh2000 committed:
A-MoreOOP: Use More OOP

amresh2000 committed:
A-Packages: Organize into Packages

amresh2000 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

amresh2000 committed:
A-JUnit: Add JUnit Tests

amresh2000 committed:
Level 9. Find

amresh2000 committed:
A-JavaDoc: JavaDoc

amresh2000 committed:
A-CodingStandard: Follow the Coding Standard

amresh2000 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard: Follow the Coding Standard

amresh2000 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  A-JavaDoc: JavaDoc

amresh2000 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level 9. Find

# Conflicts:
#	src/main/java/duke/TaskList.java

amresh2000 committed:
Added Checkstyle Files

Wu-Lezheng commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+ * Dealine Class to represent a class that will store Deadline object that extends from Tasks Class
+ * @author amresh A0235398R
+ */
+public class Deadline extends Task {
```
Maybe it be better to put classes such as Deadline.java, Event.java, etc to a separate folder called "Command" to add more clarity.

jacobkwan replied:
+1 to [Lezheng's comment](https://github.com/nus-cs2103-AY2223S1/ip/pull/237/files#r959589248), but perhaps you could name that folder "task" instead of "Command". The latter would be an appropriate folder to create if you had classes like `AddCommand`, `DeleteCommand` and other `Command` abstractions which you did not implement (and that's fine).

Wu-Lezheng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,87 @@
+            if (userInput.equals("bye")) {
+                break;
+            }
+            if (userInput.split(" ", 2)[0].equals("mark")) {
```
What about extracting the common part `userInput.split(" ", 2)[0]` in the following `if` statement into a new variable called `commandWord`?

Wu-Lezheng commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,178 @@
+public class Storage {
+
+    protected String filePath;
+    protected ArrayList<Task> storeLists = new ArrayList<>();
```
Maybe renaming `storeLists` which sounds like a verb to `storageList` or `tasks` which are nouns would be clearer to others?

Wu-Lezheng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * Method to add task to list
```
Would it be better to leave a line between `description` and `@param` in JavaDoc? I notice this issue in other methods as well.

Wu-Lezheng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+     * @param userInput Input given by user
+     * @param by Due date for the deadline
+     */
+    public void deadline(String userInput, String by) {
```
Would it be better if the method name is a verb like `createDeadline` instead of a noun like `deadline`? The same applies to other methods in this class.

Wu-Lezheng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+     * @param userInput Input given by the user
+     * @throws DukeException Error thrown that is specific to Duke
+     */
+    public void userInputCheck(String userInput) throws DukeException {
```
Would it be better to rename the method as `checkUserInput`?

Wu-Lezheng commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,178 @@
+     * Method that activates convertToDate method
+     * @param d Deadline object passed in as parameter
+     */
+    public static void dateProcessor(Deadline d) {
```
Would it be better to rename the method to `activateConversionToDate` to add more clarity in terms of what the method does?

Wu-Lezheng reviewed (commented):
Overall the code looks clean. Just a few nits to fix. Maybe the whole program could be structured in another way to have greater clarity.

jacobkwan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+
+    /**
+     * Method to convert deadline String to LocalDate format
+     * @return boolean true value if converted
```
Should there be an empty line between the description and return sections for readability? I noticed this in several of your javadocs, here's the [reference material](https://se-education.org/guides/conventions/java/intermediate.html).

jacobkwan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,41 @@
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+            LocalDate localDate = LocalDate.parse(this.by, formatter);
+            this.date = localDate;
+            this.by = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```
Perhaps you could avoid using `this` here since the variable names are not being shadowed?
```suggestion
            date = localDate;
            by = localDate.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```

jacobkwan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,178 @@
+package duke;
+
+import java.io.*;
```
Perhaps it would be better to specify exactly which imports you are using instead of using the wildcard?

amresh2000 committed:
Level 10. GUI added JavaFX

jacobkwan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,178 @@
+     * Method that parses strings and converts them to Tasks
+     * @param taskStr String that will be passed into method to convert
+     */
+    public void renderStringsAsTasks(String taskStr) {
```
A bit of a nitpick but perhaps this method could be renamed? `render` usually has other meanings (I'm thinking 3D render etc). I like your description in the accompanying javadocs, so perhaps `convertStringToTask` would be a more intuitive name? I've removed the plurality as well (`Strings` to `String`) as the method operates on one String and returns one Task. 
```suggestion
    public void convertStringToTask(String taskStr) {
```

jacobkwan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,178 @@
+    }
+
+    /**
+     * To edit Text in file primarily to mark or unmark tasks
```
Should `Text` be lower case? The implication of the capital `T` is not obvious here.

jacobkwan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,178 @@
+
+    /**
+     * To edit Text in file primarily to mark or unmark tasks
+     * @param newText New text to replace old text
+     * @param oldText Old text that is meant to be replaced
```
Should theses sentences be terminated with a full stop `.`? 
These minor nits occur in several of the javadocs

jacobkwan commented on src/test/java/DeadlineTest.java:
```diff
@@ -0,0 +1,14 @@
+public class DeadlineTest {
+
+    @Test
+    public void convertToDate_success() {
```
I like the naming of the test, its immediately clear what behaviour this test case is validating.

jacobkwan reviewed (commented):
LGTM, other than a few minor nits to fix! 👍 
Side note, PR description has some syntax errors
1. Header is included in the top quote (not sure if that's intended)
2. A space is required between the `#` and the header description
3. There should not be a space between your link description [click here]*no space here*(www.google.com) and the opening parenthesis of the actual link.

amresh2000 committed:
Remove unnecessary import statements

Reduce scope of variables to avoid common programming
errors as the project becomes bigger

Extract method to remove low level operations from main method

amresh2000 committed:
Add assert to ensure no bugs when loading task stored
in a file

amresh2000 committed:
Merge pull request #1 from amresh2000/branch-A-CodeQuality

Improve Code Quality

amresh2000 committed:
Merge branch 'master' into branch-A-Assertions

* master:
  Remove unnecessary import statements

amresh2000 committed:
Merge pull request #2 from amresh2000/branch-A-Assertions

Add assert to ensure no bugs when loading task stored

amresh2000 committed:
Add an extension: C-Detect Duplicate feature

Equals method in base and sub classes with polymorphism

amresh2000 committed:
Bug fix for remove line in file

amresh2000 committed:
Java FXML for GUI

Code Refactoring

amresh2000 committed:
A-UserGuide

Add User Guide to README.md

Add Representative Screenshot

amresh2000 committed:
Remove of unwanted text from README.md

amresh2000 committed:
Minor Code Quality Checks

amresh2000 committed:
Edit build.gradle to create JAR

amresh2000 committed:
Bug fix for end of program GUI

amresh2000 committed:
Load file bug

amresh2000 committed:
test: add test for Deadline, Event and Todo

amresh2000 committed:
bug: minor bug fix

amresh2000 committed:
docs: update UG

amresh2000 committed:
Merge pull request #5 from amresh2000/Add-test

Add test

amresh2000 committed:
style: checkStyle issues

amresh2000 committed:
Merge pull request #6 from amresh2000/CheckStyle-Edit

style: checkStyle issues
